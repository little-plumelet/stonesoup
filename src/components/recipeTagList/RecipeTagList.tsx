import { Divider } from 'antd';
import { ETags } from '../../constants';
import { IRecipeTag } from '../../interfaces/recipeTag.interface';
import { RecipeTag } from '../recipeTag';
import style from './style.module.css';

interface IRecipeTagList {
  tags: IRecipeTag[];
}

export function RecipeTagList({ tags }: IRecipeTagList) {
  const tagList: Array<JSX.Element> = [];
  tags.forEach(({ title, content }, index) => {
    if (title === ETags.servings) {
      tagList.push(
        <RecipeTag
          key={title}
          title={title}
          content={`${content}${content === 1 ? ' person' : ' people'}`}
        />
      );
    } else if (title === ETags.preptime) {
      tagList.push(
        <RecipeTag key={title} title={title} content={`${content} min`} />
      );
    } else {
      tagList.push(<RecipeTag key={title} title={title} content={content} />);
    }
    if (index !== tags.length - 1) {
      tagList.push(<Divider key={`${title}-divider`} type="vertical" />);
    }
  });

  return <div className={style.recipeTagContainer}>{tagList}</div>;
}
