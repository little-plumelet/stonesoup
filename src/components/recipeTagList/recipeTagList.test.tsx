import { render } from '@testing-library/react';
import { RecipeTagList } from './RecipeTagList';

const mockTags = [
  {
    title: 'TAG',
    content: 'tag content',
  },
  {
    title: 'REACT',
    content: 'react content',
  },
  {
    title: 'servings',
    content: 1,
  },
  {
    title: 'prep time',
    content: 34,
  },
];

describe('recipetaglist', () => {
  it('should render', () => {
    const component = render(<RecipeTagList tags={mockTags} />);
    expect(component).toMatchSnapshot();
  });
});
