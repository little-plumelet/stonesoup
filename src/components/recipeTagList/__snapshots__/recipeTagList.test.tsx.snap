// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recipetaglist should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="recipeTagContainer"
      >
        <div
          class="tagBlock"
        >
          <div
            class="tagTitle"
          >
            TAG
          </div>
          <div
            class="tagContent"
          >
            tag content
          </div>
        </div>
        <div
          class="ant-divider css-dev-only-do-not-override-mxhywb ant-divider-vertical"
          role="separator"
        />
        <div
          class="tagBlock"
        >
          <div
            class="tagTitle"
          >
            REACT
          </div>
          <div
            class="tagContent"
          >
            react content
          </div>
        </div>
        <div
          class="ant-divider css-dev-only-do-not-override-mxhywb ant-divider-vertical"
          role="separator"
        />
        <div
          class="tagBlock"
        >
          <div
            class="tagTitle"
          >
            SERVINGS
          </div>
          <div
            class="tagContent"
          >
            1 person
          </div>
        </div>
        <div
          class="ant-divider css-dev-only-do-not-override-mxhywb ant-divider-vertical"
          role="separator"
        />
        <div
          class="tagBlock"
        >
          <div
            class="tagTitle"
          >
            PREP TIME
          </div>
          <div
            class="tagContent"
          >
            34 min
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="recipeTagContainer"
    >
      <div
        class="tagBlock"
      >
        <div
          class="tagTitle"
        >
          TAG
        </div>
        <div
          class="tagContent"
        >
          tag content
        </div>
      </div>
      <div
        class="ant-divider css-dev-only-do-not-override-mxhywb ant-divider-vertical"
        role="separator"
      />
      <div
        class="tagBlock"
      >
        <div
          class="tagTitle"
        >
          REACT
        </div>
        <div
          class="tagContent"
        >
          react content
        </div>
      </div>
      <div
        class="ant-divider css-dev-only-do-not-override-mxhywb ant-divider-vertical"
        role="separator"
      />
      <div
        class="tagBlock"
      >
        <div
          class="tagTitle"
        >
          SERVINGS
        </div>
        <div
          class="tagContent"
        >
          1 person
        </div>
      </div>
      <div
        class="ant-divider css-dev-only-do-not-override-mxhywb ant-divider-vertical"
        role="separator"
      />
      <div
        class="tagBlock"
      >
        <div
          class="tagTitle"
        >
          PREP TIME
        </div>
        <div
          class="tagContent"
        >
          34 min
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
